src/data/all-moves.ts

```diff
@@ -1602,7 +1603,8 @@ export function initMoves() {
     new AttackMove(MoveId.METAL_BURST, ElementalType.STEEL, MoveCategory.PHYSICAL, -1, 100, 10, -1, 0, 4)
       .attr(
         CounterDamageAttr,
-        (move: Move) => move.category === MoveCategory.PHYSICAL || move.category === MoveCategory.SPECIAL,
+        (moveId) =>
+          allMoves[moveId].category === MoveCategory.PHYSICAL || allMoves[moveId].category === MoveCategory.SPECIAL,
```

Comment:
```suggestion
        (moveId) => allMoves[moveId].isAttackMove(),
```
I think based on your changes in the last circ dep PR this can replace the check?